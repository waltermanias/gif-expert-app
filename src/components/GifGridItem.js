import React from "react";
import PropTypes from "prop-types";

const GifGridItem = ({ title, id, url }) => {
  return (
    <div className="card animate__animated animate__bounce animate__delay-2s">
      <img src={url} alt={"imagen"} />
      <p>{title}</p>
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GifGridItem;
