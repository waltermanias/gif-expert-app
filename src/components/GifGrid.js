import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading ? "Cargando" : "Data cargada"}
      <div className="card-grid">
        {images.map((image) => {
          return <GifGridItem key={image.id} {...image} />;
        })}
      </div>
    </>
  );
};
