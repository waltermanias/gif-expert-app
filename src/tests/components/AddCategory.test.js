import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe("AddCategory", () => {
  const setCategories = jest.fn();
  const wrapper = shallow(<AddCategory setCategories={setCategories} />);

  it("should render the component", () => {
    const wrapper = shallow(<AddCategory setCategories={setCategories} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe("should not call the method", () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });
    it("should not submit any data", () => {
      wrapper.find("form").simulate("submit", { preventDefault: () => {} });
      expect(setCategories).not.toHaveBeenCalled();
    });
  });

  describe("should call the method", () => {
    it("should call setCategories method", () => {
      const input = wrapper.find("input");
      input.simulate("change", { target: { value: "Hola mundo" } });
      wrapper.find("form").simulate("submit", { preventDefault: () => {} });
      expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
      expect(input.text().trim()).toBe("");
    });
  });
});
