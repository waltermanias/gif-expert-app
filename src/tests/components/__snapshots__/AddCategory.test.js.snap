// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory should render the component 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button>
    Agregar
  </button>
</form>
`;
